File: wpilibjExamples/src/main/java/org/wpilib/templates/opmode/opmode/MyTeleop.java
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

package org.wpilib.templates.opmode.opmode;

import org.wpilib.opmode.OpMode;
import org.wpilib.opmode.Teleop;
import org.wpilib.templates.opmode.Robot;

@Teleop
public class MyTeleop implements OpMode {
  private final Robot m_robot;

  /** The Robot instance is passed into the opmode via the constructor. */
  public MyTeleop(Robot robot) {
    m_robot = robot;
  }

  @Override
  public void disabledPeriodic() {
    /* Called periodically (on every DS packet) while the robot is disabled. */
  }

  @Override
  public void opModeStart() {
    /* Called once when the robot is enabled. */
  }

  @Override
  public void opModePeriodic() {
    /* Called periodically (set time interval) while the robot is enabled. */
  }

  @Override
  public void opModeStop() {
    /* Called when the robot is disabled (after previously being enabled). */
  }

  @Override
  public void opModeClose() {
    /* Called when the opmode is de-selected / no additional methods will be called. */
  }
}

File: wpilibcExamples/src/main/cpp/templates/opmode/cpp/opmode/MyTeleop.cpp
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

#include "opmode/MyTeleop.hpp"

#include "Robot.hpp"

/** The Robot instance is passed into the opmode via the constructor. */
MyTeleop::MyTeleop(Robot& robot) : m_robot{robot} {}

MyTeleop::~MyTeleop() {
  /* Called when the opmode is de-selected. */
}

void MyTeleop::Start() {
  /* Called once when the robot is first enabled. */
}

void MyTeleop::Periodic() {
  /* Called periodically (set time interval) while the robot is enabled. */
}

void MyTeleop::End() {
  /* Called when the robot is disabled (after previously being enabled). */
}
